pkg/skaffold/util/tar.go

```diff
@@ -62,109 +86,81 @@ func CreateTarGz(w io.Writer, root string, paths []string) error {
 	return CreateTar(gw, root, paths)
 }
 
-func addFileToTar(root string, src string, dst string, tw *tar.Writer) error {
-	var (
-		absPath string
-		err     error
-	)
-
-	absRoot, err := filepath.Abs(root)
+func addFileToTar(root string, src string, dst string, tw *tar.Writer, hm headerModifier) error {
+	fi, err := os.Lstat(src)
 	if err != nil {
 		return err
 	}
 
-	if filepath.IsAbs(src) {
-		absPath = src
-	} else {
-		absPath, err = filepath.Abs(src)
-		if err != nil {
-			return err
-		}
+	mode := fi.Mode()
+	if mode&os.ModeSocket != 0 {
+		return nil
 	}
 
-	tarPath := dst
-	if tarPath == "" {
-		tarPath, err = filepath.Rel(absRoot, absPath)
+	var header *tar.Header
+	if mode&os.ModeSymlink != 0 {
+		target, err := os.Readlink(src)
 		if err != nil {
 			return err
 		}
-	}
-	tarPath = filepath.ToSlash(tarPath)
 
-	fi, err := os.Lstat(absPath)
-	if err != nil {
-		return err
-	}
-	switch mode := fi.Mode(); {
-	case mode.IsDir():
-		tarHeader, err := tar.FileInfoHeader(fi, tarPath)
-		if err != nil {
-			return err
+		if filepath.IsAbs(target) {
```

Comment:
Should we further prevent references that resolve outside of the `context` directory?